### PR TITLE
Let travis check for git format mistakes

### PR DESCRIPTION
**Pull request by est31**
With this PR, travis checks now for multiple types of bad git formatting.

### CODE REVIEW

**Comment by est31** [comment]
I've added some failing commits to show what travis will print.

**Comment by ShadowNinja** [comment]
Should fail for the Windows builds too.

**Comment by est31** [comment]
@ShadowNinja done.

File: util/travis/script.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ if [[ $PLATFORM == "Linux" ]]; then
 	cmake $CMAKE_FLAGS ..
 	make -j2
 	echo "Running unit tests."
-	../bin/minetest --run-unittests && exit 0
+	../bin/minetest --run-unittests || exit 1
```
Comment (sfan5) [inline util/travis/script.sh]: actually you can drop this because we call bash with the -e flag

File: util/travis/script.sh [hunk 2]
```diff
@@ -31,12 +31,14 @@ elif [[ $PLATFORM == Win* ]]; then
 	export EXISTING_MINETEST_DIR=$OLDDIR
 	export NO_MINETEST_GAME=1
 	if [[ $PLATFORM == "Win32" ]]; then
-		$OLDDIR/util/buildbot/buildwin32.sh travisbuild && exit 0
+		$OLDDIR/util/buildbot/buildwin32.sh travisbuild || exit 1
 	elif [[ $PLATFORM == "Win64" ]]; then
-		$OLDDIR/util/buildbot/buildwin64.sh travisbuild && exit 0
+		$OLDDIR/util/buildbot/buildwin64.sh travisbuild || exit 1
```
Comment (sfan5) [inline util/travis/script.sh]: same with these

File: util/travis/script.sh [hunk 3]
```diff
@@ -31,12 +31,14 @@ elif [[ $PLATFORM == Win* ]]; then
...
 	fi
 else
 	echo "Unknown platform \"${PLATFORM}\"."
 	exit 1
 fi
 
+echo "Checking commits."
+../util/check_git_guidelines.sh && exit 0
```
Comment (sfan5) [inline util/travis/script.sh]: this doesn't work: https://travis-ci.org/minetest/minetest/jobs/78973734#L5832

**Pull request closed without merging**